statics/src/main/java/com/codeborne/selenide/impl/WebDriverThreadLocalContainer.java

```diff
@@ -124,11 +124,17 @@ else if (webDriver == null) {
 
   private WebDriver createDriver() {
     CreateDriverCommand.Result result = createDriverCommand.createDriver(config, factory, userProvidedProxy, listeners);
-    threadWebDriver.put(currentThread().getId(), result.webDriver);
+    long threadId = currentThread().getId();
+    threadWebDriver.put(threadId, result.webDriver);
     if (result.selenideProxyServer != null) {
-      threadProxyServer.put(currentThread().getId(), result.selenideProxyServer);
+      threadProxyServer.put(threadId, result.selenideProxyServer);
+    }
+    if (config.holdBrowserOpen()) {
+      log.info("Hold browser and proxy open: {} -> {}, {}", threadId, result.webDriver, result.selenideProxyServer);
```

Comment:
Will it be good to have duplicated log messages here and [CloseDriverCommand:18](https://github.com/selenide/selenide/blob/master/src/main/java/com/codeborne/selenide/drivercommands/CloseDriverCommand.java#L18)? Maybe at this point, we can say that browser will not be closed after running? What do you think?